SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -455,7 +437,77 @@ void InterestFactory::AddComponentQueryPairToInterestComponent(Interest& OutInte
 	OutInterest.ComponentInterestMap[ComponentId].Queries.Add(QueryToAdd);
 }
 
-void InterestFactory::GetActorUserDefinedQueries(const AActor* InActor, const QueryConstraint& LevelConstraints, TArray<SpatialGDK::Query>& OutQueries, bool bRecurseChildren) const
+TArray<Query> InterestFactory::GetUserDefinedQueries(const AActor* InActor, const QueryConstraint LevelConstraint) const
+{
+	SCOPE_CYCLE_COUNTER(STAT_InterestFactoryAddUserDefinedQueries);
+
+	TMap<float, TArray<QueryConstraint>> FrequencyToConstraintsMap = GetUserDefinedFrequencyToConstraintsMap(InActor);
+	TArray<Query> Queries;
+
+	for (const auto& FrequencyToConstraints : FrequencyToConstraintsMap)
+	{
+		Query UserQuery;
+		QueryConstraint UserConstraint;
+		// Assign this query a frequency if it's specified. Otherwise leave it empty for no rate limiting.
+		if (FrequencyToConstraints.Key != 0)
+		{
+			UserQuery.Frequency = FrequencyToConstraints.Key;
```

Comment:
So UserQuery.Frequency is 0 by default?
If so can we just remove the if statement and have: UserQuery.Frequency = FrequencyToConstraints.Key;

Reply:
I wrote it this way to be explicit. It is redundant to have the if statement though, so will remove.